src/main/scala/device/RocketDebugWrapper.scala

```diff
@@ -112,34 +111,33 @@ object XSDebugModuleParams {
 
 case object EnableJtag extends Field[Bool]
 
-class SimJTAG(tickDelay: Int = 50)(implicit val p: Parameters) extends BlackBox(Map("TICK_DELAY" -> IntParam(tickDelay)))
-  with HasBlackBoxResource {
-  val io = IO(new Bundle { 
-    val clock = Input(Clock())
-    val reset = Input(Bool())
-    val jtag = new JTAGIO(hasTRSTn = true)
-    val enable = Input(Bool())
-    val init_done = Input(Bool())
-    val exit = Output(UInt(32.W))
-  })
+class SimJTAG(tickDelay: Int = 50)(implicit val p: Parameters) extends ExtModule(Map("TICK_DELAY" -> IntParam(tickDelay)))
+  with HasExtModuleResource {
+
+  val clock = IO(Input(Clock()))
+  val reset = IO(Input(Bool()))
+  val jtag = IO(new JTAGIO(hasTRSTn = true))
+  val enable = IO(Input(Bool()))
+  val init_done = IO(Input(Bool()))
+  val exit = IO(Output(UInt(32.W)))
 
   def connect(dutio: JTAGIO, tbclock: Clock, tbreset: Bool, init_done: Bool, tbsuccess: Bool) = {
-    dutio.TCK := io.jtag.TCK
-    dutio.TMS := io.jtag.TMS
-    dutio.TDI := io.jtag.TDI
-    io.jtag.TDO := dutio.TDO
+    dutio.TCK := jtag.TCK
+    dutio.TMS := jtag.TMS
+    dutio.TDI := jtag.TDI
+    jtag.TDO := dutio.TDO
 
-    io.clock := tbclock
-    io.reset := tbreset
+    clock := tbclock
+    reset := tbreset
 
-    io.enable    := p(EnableJtag)
-    io.init_done := init_done
+    enable    := p(EnableJtag)
+    init_done := init_done
```

Comment:
conflict between IO ports and function arguments

Reply:
WTF, I'll resolve this tonight.